src/main/java/com/avail/optimizer/L2ValueManifest.java

```diff
@@ -1332,4 +1359,24 @@ public boolean forgetRegisters (final Set<L2Register> registersToForget)
 		});
 		return anyChanged.value;
 	}
+
+	/**
+	 * Forget only the {@link RegisterKind#BOXED} register definitions for this
+	 * synonym.  Keep the synonym around in all circumstances, even if there are
+	 * no remaining definitions.  Also, do not alter the synonym's restriction.
```

Comment:
In context of this review, I holistically understand what this is for and why, but a general traversal of the code will find the narrative more obscure. Probably best to explain here that `makeImmutable` needs this, and how else it might be used in the future.

Reply:
Done.